include/taskolib/Step.h

```diff
@@ -304,6 +315,18 @@ class Step
     bool is_running_{ false };
     bool is_disabled_{ false };
 
+    /**
+     * Checks if the step can execute a setup script before the appended intrinsic script.
+     *
+     * \return true allow to execute the setup script
+     * \return false refuse on executing the setup script
+     */
+    bool allow_step_setup() const
+    {
+        return get_type() == type_action or get_type() == type_if
+            or get_type() == type_elseif or get_type() == type_while;
+    }
```

Comment:
Oh... do we want this? An END can rightfully have a script, I am just not sure if it is ever executed. We should make sure that the setup is run whenever a Step script is run.

Reply:
I can not find, but I remember that we discussed a function or Step object member that contained/detected just this set of step types: Types that are really executed.

I would rather see that reborn / dug out then to have this here as list. It is much more declarative to set that somewhere and just utilize the property this-is-an-executable-step here.

And I'm sure it is useful. We needed it before, we need it now, we will need it in the future.

So the function / property of Step would not be `allow_step_setup` but `step_executes_a_script` (or whatever, I believe we had a nice name). That is much more general and describes what we want to do here far better.

Reply:
Compare:

3b8cef3f `has_lua_script()` (meanwhile removed with a9167fd8af)

b0c659e272 `is_continuation_type()` (removed with 0d75b741c0)

`requires_bool_return_value()` is somehow also a property of the `Step::Type` modelled as property of `Step`.

For me `has_lua_script()` (or rather `has_script()`) is a better name for this, more general and describes why we want to enable the setup-script; the solution choosen here is technically correct/identical but does not transport the real reason.

Additionally: I would unify the 'property of a Step (-Type)' functions somehow, and make them public. As we discussed (on Gitlab, around b0c659e272 I guess) I believe the properties should be thought of as 'first class citizens' and not 'some byproduct we ignore'. They are what would make metaprogramming possible later. At the moment the code looks in some places like 'everything is spelled out in minute detail' rather than take a step back and show the greater pattern.
Some of the `case` cascades on `Step::Type` could reveal more of the underlying ideas if we would have `Step::Type` properties that we could utilize.
This has nothing to do (at least not very strict) with this PR, it just comes up in my head every time I look on the code, and it came up here. Maybe something to let stew in the back of our heads.

Reply:
From the other 'property functions' in `Step`

```c++
    bool is_running() const;
    bool is_disabled() const;
    bool is_executing_scripts() const;
```

Or something?

Reply:
Just one comment: has_script() is a misleading name because every Step has a script member. We can even legally change the type of a Step from action to end and the script remains there.

Reply:
Maybe `executes_script()` would be an appropriate name for this function?

Reply:
Changed.